app/helpers/search_helper.rb

```diff
@@ -66,5 +66,14 @@ def assign_material_link(material, action, extra={})
       link_to("Assign to a Class", action, extra.merge({:class=>"button"}))
     end
   end
-
+  
+  # return the grade_level_groups for which materials are authored
+  def authored_grade_level_groupes
```

Comment:
@scytacki @knowuh Can this be the method which is slowing down the search page?
This method contains implementation for displaying only those grade levels check boxes for which there is a material authored in the filter section. Subject areas too uses similar logic to display check boxes.